components/tag-list/tag-list.js

```diff
@@ -228,19 +230,34 @@ class TagList extends LocalizeCoreElement(ArrowKeysMixin(LitElement)) {
 	}
 
 	_handleResize(entries) {
-		this._availableWidth = Math.ceil(entries[0].contentRect.width);
+		this._availableWidth = Math.floor(entries[0].contentRect.width);
 		if (this._availableWidth >= PAGE_SIZE.large) this._lines = PAGE_SIZE_LINES.large;
 		else if (this._availableWidth < PAGE_SIZE.large && this._availableWidth >= PAGE_SIZE.medium) this._lines = PAGE_SIZE_LINES.medium;
 		else this._lines = PAGE_SIZE_LINES.small;
+		if (!this._hasResized) {
+			this._hasResized = true;
+			this._handleSlotChange();
+		}
 		this._chomp();
 	}
 
 	_handleSlotChange() {
-		requestAnimationFrame(() => {
+		if (!this._hasResized) return;
+
+		requestAnimationFrame(async() => {
 			this._items = this._getTagListItems();
-			this._itemLayouts = this._getItemLayouts(this._items);
+			if (!this._items || this._items.length === 0) return;
+
+			const numItems = this._items.length;
```

Comment:
Note: [forEach expects a synchronous function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach).  It does not wait for promises.  It might seem like it's working, but I don't think it's doing what we think it is. Because it's not actually waiting, there is no guarantee that the previous items are actually done updating when the last item update completes.

To wait in sequence, I think it is possible to use `for...of` loop, or _maybe_ it's `for await...of`.  However a better approach is to wait in parallel by mapping to an array of promises.  This example uses `map`, but they could be manually collected in an array. 

```javascript
await Promise.all(this._items.map(item => item.updateComplete));
```